### PR TITLE
feat(IDX): use env vars for embedders test deps

### PR DESCRIPTION
**Pull request by nmattia**
This changes the embedders spec tests to use environment variables (set in `BUILD.bazel`) to specify the path to the wasm spec files.

This ensures that the tests do not depend on Bazel paths like `external/` and (more theoretically than practically) will make the tests run only if a file specifically depended on changes.

### CODE REVIEW

**Review by basvandijk — APPROVED** [review]

File: WORKSPACE.bazel [hunk 1]
```diff
@@ -328,11 +328,26 @@ filegroup(
...
+filegroup(
+    name = "multi_memory_wast_files",
+    srcs = glob(["proposals/multi-memory/*.wast"]),
+    visibility = ["//visibility:public"]
+)
+filegroup(
+    name = "memory64_wast_files",
+    srcs = glob(["proposals/memory64/*.wast"]),
+    visibility = ["//visibility:public"]
+)
```
Comment (adambratschikaye) [inline WORKSPACE.bazel:348]: Should these be indented since they're under `http_archive(...`?

**Review by adambratschikaye — APPROVED** [review]
Thanks!

File: rs/embedders/tests/spec_tests.rs [hunk 2]
```diff
@@ -794,29 +799,24 @@ fn error_to_string(e: anyhow::Error) -> String {
 /// included in our repo, but is imported by Bazel using the `new_git_repository`
 /// rule in `WORKSPACE.bazel`.
 ///
-/// If you need to look at the test `wast` files directly they can be found in
+/// See BUILD.bazel for inspecting the `wast` files.
 /// `bazel-ic/external/wasm_spec_testsuite/` after building this test.
 #[test]
```
[the remark this answers is not in the document]
Comment (nmattia) [reply rs/embedders/tests/spec_tests.rs:804]: Good catch!

File: rs/embedders/tests/spec_tests.rs [hunk 3]
```diff
@@ -794,29 +799,24 @@ fn error_to_string(e: anyhow::Error) -> String {
 /// included in our repo, but is imported by Bazel using the `new_git_repository`
 /// rule in `WORKSPACE.bazel`.
 ///
-/// If you need to look at the test `wast` files directly they can be found in
+/// See BUILD.bazel for inspecting the `wast` files.
 /// `bazel-ic/external/wasm_spec_testsuite/` after building this test.
```
Comment (nmattia) [inline rs/embedders/tests/spec_tests.rs:803]:
```suggestion
```

File: WORKSPACE.bazel (hunk 1 above)
Comment (nmattia) [reply WORKSPACE.bazel:348]: I think it doesn't matter too much; we do a bit of a mix of both

**Pull request merged**